python_modules/dagster/dagster/_core/definitions/instigation_logger.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from contextlib import ExitStack
+from typing import IO, List, Optional
+
+from dagster import _seven
+from dagster._core.instance import DagsterInstance
+from dagster._core.storage.captured_log_manager import CapturedLogManager
+from dagster._core.utils import coerce_valid_log_level
+from dagster._utils.log import create_console_logger
+
+
+class InstigationLogHandler(logging.Handler):
+    """
+    The main purpose of this log handler is to proxy logging records to a set of downstream loggers
+    which themselves will route to their own set of handlers
+    """
+
+    def __init__(self, downstream_loggers: List[logging.Logger]):
+        self._should_capture = True
+        self._downstream_loggers = [*downstream_loggers]
+        super().__init__()
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        return self._should_capture
+
+    def emit(self, record: logging.LogRecord):
+        """For any received record, add metadata, and have handlers handle it"""
+
+        try:
+            self._should_capture = False
+            for logger in self._downstream_loggers:
+                logger.handle(record)
+        finally:
+            self._should_capture = True
+
+
+class CapturedLogHandler(logging.Handler):
```

Comment:
Functionally, this is basically writing log records as JSON to a stream, right? Similar to above feedback, a name that reflects that might make this more clear.